plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/UsageAwareObjectDescription.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.scoping;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.xtext.naming.QualifiedName;
+import org.eclipse.xtext.resource.IEObjectDescription;
+
+/**
+ * An {@link IEObjectDescription} that has the callback-method {@link #markAsUsed()} that is invoked when this object
+ * description is used to bind a name.
+ */
+public abstract class UsageAwareObjectDescription<T extends IEObjectDescription> implements IEObjectDescription {
```

Comment:
1. the "E" in front of "Object" is missing in the name, here.
2. Interface `UsageAwareEObjectDescription` should then be renamed to `IUsageAwareEObjectDescription` (note the "I" at the beginning).